server/src/main/java/org/apache/kafka/server/config/KafkaLogConfigs.java

```diff
@@ -0,0 +1,166 @@
+            "this specified threshold. This configuration is ignored if log.message.timestamp.type=LogAppendTime.";
+
+    public static final String NUM_RECOVERY_THREADS_PER_DATA_DIR_CONFIG = "num.recovery.threads.per.data.dir";
+    public static final int NUM_RECOVERY_THREADS_PER_DATA_DIR_DEFAULT = 1;
+    public static final String NUM_RECOVERY_THREADS_PER_DATA_DIR_DOC = "The number of threads per data directory to be used for log recovery at startup and flushing at shutdown";
+
+    public static final String AUTO_CREATE_TOPICS_ENABLE_CONFIG = "auto.create.topics.enable";
+    public static final boolean AUTO_CREATE_TOPICS_ENABLE_DEFAULT = true;
+    public static final String AUTO_CREATE_TOPICS_ENABLE_DOC = "Enable auto creation of topic on the server.";
+
+    public static final String MIN_IN_SYNC_REPLICAS_CONFIG = ServerTopicConfigSynonyms.serverSynonym(TopicConfig.MIN_IN_SYNC_REPLICAS_CONFIG);
+    public static final String MIN_IN_SYNC_REPLICAS_DOC = "When a producer sets acks to \"all\" (or \"-1\"), " +
+            "<code>min.insync.replicas</code> specifies the minimum number of replicas that must acknowledge " +
+            "a write for the write to be considered successful. If this minimum cannot be met, " +
+            "then the producer will raise an exception (either <code>NotEnoughReplicas</code> or " +
+            "<code>NotEnoughReplicasAfterAppend</code>).<br>When used together, <code>min.insync.replicas</code> and acks " +
+            "allow you to enforce greater durability guarantees. A typical scenario would be to " +
+            "create a topic with a replication factor of 3, set <code>min.insync.replicas</code> to 2, and " +
+            "produce with acks of \"all\". This will ensure that the producer raises an exception " +
+            "if a majority of replicas do not receive a write.";
+
+    public static final String CREATE_TOPIC_POLICY_CLASS_NAME_CONFIG = "create.topic.policy.class.name";
+    public static final String CREATE_TOPIC_POLICY_CLASS_NAME_DOC = "The create topic policy class that should be used for validation. The class should " +
+            "implement the <code>org.apache.kafka.server.policy.CreateTopicPolicy</code> interface.";
+    public static final String ALTER_CONFIG_POLICY_CLASS_NAME_CONFIG = "alter.config.policy.class.name";
+    public static final String ALTER_CONFIG_POLICY_CLASS_NAME_DOC = "The alter configs policy class that should be used for validation. The class should " +
+            "implement the <code>org.apache.kafka.server.policy.AlterConfigPolicy</code> interface.";
+    public static final String LOG_MESSAGE_DOWNCONVERSION_ENABLE_CONFIG = ServerTopicConfigSynonyms.serverSynonym(TopicConfig.MESSAGE_DOWNCONVERSION_ENABLE_CONFIG);
+    public static final String LOG_MESSAGE_DOWN_CONVERSION_ENABLE_DOC = TopicConfig.MESSAGE_DOWNCONVERSION_ENABLE_DOC;
```

Comment:
It seems to me we should make consistency for all docs here. 1) make it point to `TopicConfig.xxxDOC` or 2) copy the docs to here.

Reply:
I don't think we can copy all configs doc to `TopicConfig` as the docs are different in many places between topic level config and topic broker level config. I will copy it for now into KafkaLogConfigs in case the doc diverted in the futrue like the rest of the docs KafkaLogConfigs.